Add warning when multi-thread decompression is requested

tomcwang opened the pull request:
address https://github.com/facebook/zstd/issues/2918

When user pass in argument for both decompression and multi-thread, print a warning message
to indicate that multi-threaded decompression is not supported.

tomcwang committed:
Add warning when multi-thread decompression is requested

When user pass in argument for both decompression and multi-thread, print a warning message
to indicate that multi-threaded decompression is not supported.

yoniko commented:
Thank you @tomcwang .
I see the check is inserted after `-T0` is handled.
That means that a warning will be printed on decompression with `-T0` on a multicore system.
Can you move the check to before the `-T0` case is handled and `nbWorkers` are assigned?

tomcwang committed:
Move check to before nbWorker is assigned

move check to before nbWorker is assigned so we do not throw warning message when
-T0 is used.

tomcwang commented:
> Thank you @tomcwang . I see the check is inserted after `-T0` is handled. That means that a warning will be printed on decompression with `-T0` on a multicore system. Can you move the check to before the `-T0` case is handled and `nbWorkers` are assigned?

makes sense, I moved the check to before nbWorker is assigned, warning is not thrown for -T0 anymore.

yoniko commented:
Thank you.
One last thing - can you please add tests to `cli-tests/compression/multi-threaded.sh`?

tomcwang committed:
add test case for multi-threaded decoding warning

add test case for multi-threaded decoding warning.
Expecation is for -d -T0 we will not throw any warning,
and see warning for any other -d -T(>1) inputs

yoniko reviewed (approved):
Looks good!
